File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.py
class Solution:
    def searchMatrix(self, matrix: List[List[int]], target: int) -> bool:
        r_max, c_max = len(matrix), len(matrix[0])
        
        row, column = len(matrix) - 1 , 0
        
        while 0 <= row < r_max and 0 <= column < c_max:
            current = matrix[row][column]
            
            if target == current:
                return True
            elif target < current:
                row -= 1
            else:
                column += 1
        return False

File: 240-search-a-2d-matrix-ii/240-search-a-2d-matrix-ii.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        
        int r_max = matrix.size();
        int c_max = matrix[0].size();
            
        int row = r_max - 1;
        int col = 0;
        int current;
        
        while ((0<=row) && (col<c_max)){
            
            current = matrix[row][col];
            
            if (current == target)
                return true;
            else if (current < target)
                col++;
            else
                row--;
        }
        
        return false;
    }
};